azurelinuxagent/ga/exthandlers.py

```diff
@@ -214,20 +214,25 @@ class ExtHandlersHandler(object):
     def __init__(self):
         self.protocol_util = get_protocol_util()
         self.protocol = None
+        self.ext_conf = None
         self.ext_handlers = None
+        self.last_instantiation = None
+        self.artifacts_profile = None
+        self.last_vm_artifacts_seqno = None
         self.last_etag = None
         self.log_report = False
-        self.log_etag = True
+        self.log_instantiation = True
         self.log_process = False
 
         self.report_status_error_state = ErrorState()
         self.get_artifact_error_state = ErrorState(min_timedelta=ERROR_STATE_DELTA_INSTALL)
 
     def run(self):
-        self.ext_handlers, etag = None, None
+        self.ext_handlers, instantiation = None, None
         try:
             self.protocol = self.protocol_util.get_protocol()
-            self.ext_handlers, etag = self.protocol.get_ext_handlers()
+            self.ext_conf, instantiation = self.protocol.get_ext_conf()
+            self.ext_handlers = self.ext_conf.ext_handlers
```

Comment:
Given a choice, I would put in a method call here to a method which readies all the things needed to check if the fast track things are in place or not. Rather than dividing that piece into `get_artifacts_profile`, `get_sequence_number` and `process_vm_artifacts_profile_for_fast_track`. Most of the things done in `process_vm_artifacts_profile_for_fast_track` could have been done beforehand.

If your concern was the unnecessary processing done by us if there are no new settings coming in, then we can discuss and find a better way.